app/controllers/CheckYourAnswersController.scala

```diff
@@ -43,9 +46,14 @@ class CheckYourAnswersController @Inject() (
     Ok(view(list))
   }
 
-  // TODO replace dummy route and post session data
-  def onSubmit: Action[AnyContent] = authorise { implicit request =>
-    Redirect(routes.DummyController.onPageLoad.url)
+  def onSubmit: Action[AnyContent] = (authorise andThen getData).async { implicit request =>
+    routerService
+      .setUpProfile(request.eori, request.userAnswers.maintainProfileAnswers)
+      .fold(
+        error => Redirect(routes.DummyController.onPageLoad.url), //TODO redirect to actual error page
+        success => Redirect(routes.HomepageController.onPageLoad.url)
+      )(catsStdInstancesForFuture(defaultExecutionContext))
```

Comment:
Is the catsStdInstancesForFuture used for implicit excution context? I'm used to seeing `class XyzClass @Inject() ( <some injects>) (implicit ec ExecutionContext)`. I'm not averse to this cats way if we know it works properly.

Reply:
nah so it wants an implicit Functor[Future], which is the catsStdInstancesForFuture, which then itself wanted an execution context.

Have moved it up out of the method call and it looks a bit nicer I think. Let me know if that is clearer

Reply:
That's clearer :)